src/screencast/pipewire_screencast.c

```diff
@@ -26,19 +26,28 @@ static void writeFrameData(void *pwFramePointer, void *wlrFramePointer,
 	return;
 }
 
-static inline struct spa_pod *build_format(struct spa_pod_builder *b, uint32_t format, uint32_t format_without_alpha,
+static inline struct spa_pod *build_format(struct spa_pod_builder *b, uint32_t format,
 		uint32_t width, uint32_t height, uint32_t framerate)
 {
-	struct spa_pod_frame f;
-	spa_pod_builder_push_object(b, &f, SPA_TYPE_OBJECT_Format, SPA_PARAM_EnumFormat);
+	struct spa_pod_frame f[2];
+
+	enum spa_video_format format_without_alpha = xdpw_format_pw_strip_alpha(format);
+
+	spa_pod_builder_push_object(b, &f[0], SPA_TYPE_OBJECT_Format, SPA_PARAM_EnumFormat);
 	spa_pod_builder_add(b, SPA_FORMAT_mediaType, SPA_POD_Id(SPA_MEDIA_TYPE_video), 0);
 	spa_pod_builder_add(b, SPA_FORMAT_mediaSubtype, SPA_POD_Id(SPA_MEDIA_SUBTYPE_raw), 0);
-	if (format_without_alpha != SPA_VIDEO_FORMAT_UNKNOWN) {
-		spa_pod_builder_add(b, SPA_FORMAT_VIDEO_format,
-			SPA_POD_CHOICE_ENUM_Id(3, format, format, format_without_alpha), 0);
+	/* format */
+	spa_pod_builder_prop(b, SPA_FORMAT_VIDEO_format, 0);
```

Comment:
IMHO this logic is less readable than the original one.

Reply:
I will do a similar logic for modifiers so i wanted to make things similar, but we can just use the old logic.

Reply:
added a commit which simplifies the path with alpha. we can squash them, but i didn't want to loose the old version for now.